src/js/game/systems/item_ejector.js

```diff
@@ -159,17 +159,28 @@ export class ItemEjectorSystem extends GameSystemWithFilter {
                     continue;
                 }
 
+                let maxProgress = 1.0;
+
+                // Check if there's an item blocking the ejector
+                const destPath = sourceSlot.cachedBeltPath;
+                if (destPath) {
+                    maxProgress = Math.min(
+                        1.0,
+                        sourceSlot.cachedBeltPath.spacingToFirstItem / globalConfig.itemSpacingOnBelts

```

Comment:
This is nitpicking, but maybe use `destPath` here too, just for clarity?

Reply:
not nit-picky at all, I totally just missed that lol